nanoPU_sim.py

```diff
@@ -242,8 +275,14 @@ def getRxMsgID(self, src_ip, src_context, tx_msg_id, msg_len):
         self.log('Processing getRxMsgID extern call for: {}'.format(key))
         # check if this msg has already been allocated an rx_msg_id
         if key in self.rx_msg_id_table:
-            self.log('Found rx_msg_id: {}'.format(self.rx_msg_id_table[key]))
-            return self.rx_msg_id_table[key]
+            rx_msg_id = self.rx_msg_id_table[key]
+            self.log('Found rx_msg_id: {}'.format(rx_msg_id))
+            # compute the beginning of the inflight window
+            ack_no = find_first_one(~self.received_bitmap[rx_msg_id])
```

Comment:
Interesting. So you're saying that the receiver doesn't need to maintain a separate `pullOffset` state variable because the pull offset can be computed from the `received_bitmap` state variable?

Reply:
Yes. 
To be honest, someone may need to keep such a state variable when implementing their own protocol in the future but I couldn't think of any use-case for now.